src/cargo/util/config/mod.rs

```diff
@@ -2117,6 +2117,9 @@ pub struct CargoHttpConfig {
     pub debug: Option<bool>,
     pub multiplexing: Option<bool>,
     pub ssl_version: Option<SslVersionConfig>,
+    pub client_ssl_cert: Option<ConfigRelativePath>,
+    pub client_ssl_key: Option<ConfigRelativePath>,
+    pub client_ssl_key_password: Option<String>,
```

Comment:
I didn't see a linked issue or discussion.  We are wanting to discuss problems and solutions in Issues before moving on to implementation in PRs.  This helps make sure we are all on the same page and reduces the pressure on the tight resources of the cargo team.  For example, I would want more experienced people than I involved in the conversation about how the password should be handled to be both usable and secure.

> We encourage people to discuss their design before hacking on code. This gives the Cargo team a chance to know your idea more. Sometimes after a discussion, we even find a way to solve the problem without coding! Typically, you [file an issue](https://github.com/rust-lang/cargo/issues) or start a thread on the [internals forum](https://internals.rust-lang.org/c/tools-and-infrastructure/cargo) before submitting a pull request. Please read [the process](https://doc.crates.io/contrib/process/index.html) of how features and bugs are managed in Cargo.

https://doc.crates.io/contrib/process/working-on-cargo.html#before-hacking-on-cargo

See also https://blog.rust-lang.org/inside-rust/2022/03/31/cargo-team-changes.html

Reply:
I'd also encourage following the PR template as that covers things like what the testing strategy for the implementation looks like.

Reply:
I made an issue explaining the design decisions, but to be honest the password one is impossible to solve without building a proper password manager in cargo. 
Let me just point out that registry credentials are stored in plain text as well.